services/sqs/src/test/java/software/amazon/awssdk/services/sqs/SqsBatchManagerTest.java

```diff
@@ -0,0 +1,391 @@
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.batchmanager.BatchManager;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sqs.batchmanager.SqsBatchManager;
+import software.amazon.awssdk.services.sqs.internal.batchmanager.DefaultSqsBatchManager;
+import software.amazon.awssdk.services.sqs.model.ChangeMessageVisibilityBatchResponse;
+import software.amazon.awssdk.services.sqs.model.ChangeMessageVisibilityRequest;
+import software.amazon.awssdk.services.sqs.model.ChangeMessageVisibilityResponse;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchResponse;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageRequest;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageResponse;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchResponse;
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
+import software.amazon.awssdk.services.sqs.model.SendMessageResponse;
+import software.amazon.awssdk.services.sqs.model.SqsException;
+import software.amazon.awssdk.utils.BinaryUtils;
+import software.amazon.awssdk.utils.Md5Utils;
+import software.amazon.awssdk.utils.ThreadFactoryBuilder;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SqsBatchManagerTest {
+
+    private static final int DEFAULT_MAX_BATCH_OPEN = 200;
+    private static final String DEFAULT_QUEUE_URl = "SomeQueueUrl";
+    private static final URI HTTP_LOCALHOST_URI = URI.create("http://localhost:8080/");
+    private static ScheduledExecutorService scheduledExecutor;
+    private static ExecutorService executor;
+    private static SqsClient client;
+    private SqsBatchManager batchManager;
+
+    @Mock
+    private BatchManager<SendMessageRequest, SendMessageResponse, SendMessageBatchResponse> mockSendMessageBatchManager;
+
+    @Mock
+    private BatchManager<DeleteMessageRequest, DeleteMessageResponse, DeleteMessageBatchResponse> mockDeleteMessageBatchManager;
+
+    @Mock
+    private BatchManager<ChangeMessageVisibilityRequest, ChangeMessageVisibilityResponse,
+        ChangeMessageVisibilityBatchResponse> mockChangeVisibilityBatchManager;
+
+    @Rule
+    public WireMockRule wireMock = new WireMockRule();
```

Comment:
We should use `new WireMockRule(0)` to let it automatically find a usable port instead of using 8080

https://github.com/aws/aws-sdk-java-v2/blob/master/test/protocol-tests/src/test/java/software/amazon/awssdk/protocol/tests/timeout/sync/SyncTimeoutTests.java#L45